server/py/framework/rundb/sqldb.py

```diff
@@ -1304,6 +1305,22 @@ def list_alert_activations(
     ):
         raise NotImplementedError
 
+    def update_alert_activation(
+        self,
+        activation_id: int,
+        activation_time: datetime.datetime,
+        notifications_states,
+    ):
+        # We run this function with a new session because it may run concurrently.
+        # Older sessions will not be able to see the changes made by this function until they are committed.
+        return self._transform_db_error(
+            framework.db.session.run_function_with_new_db_session,
+            services.alerts.crud.AlertActivation().update_alert_activation,
+            activation_id=activation_id,
+            activation_time=activation_time,
+            notifications_states=notifications_states,
+        )
```

Comment:
In what flow is this DB class instance called and not the `sqldb/db.py` class?
The notification pusher is on the server-side, so doesn't it has access to the sql db class?

Reply:
@TomerShor 
> The notification pusher is on the server-side, so doesn't it has access to the sql db class?

this one yes, but 1- it inherits the base class which can work in [both client and server](https://github.com/mlrun/mlrun/blob/da480edda6b9a8a09e7211260b1f86e673a95038/mlrun/utils/notifications/notification_pusher.py#L45-L70) and 2- I tried to use normal sql class and got circular imports problem 3- in all the places in notification pusher this proxy class is used